バグ: E2E全テスト実行時に next/headers module resolution エラーでdevサーバー起動失敗

## 概要

`npx playwright test`（全テスト実行）で dev サーバーが起動に失敗する。個別テストファイル指定（`npx playwright test e2e/xxx.spec.ts`）では正常に動作する。

## エラー内容

```
Error: Cannot find module '/home/.../node_modules/next/headers' imported from /home/.../src/lib/server-auth.ts
Did you mean to import "next/headers.js"?
```

## 再現手順

1. `npx playwright test` を実行
2. WebServer 起動中に上記エラーが発生しテスト実行不可

## 影響

- **master ブランチでも再現する**（特定の変更に起因しない既存問題）
- pre-commit フックの `e2e-test` ステップが常に失敗するため、コミット時に `SKIP=e2e-test` が必要
- 個別テストファイル指定では問題ないため、実際のテスト自体は動作する

## 原因推測

Next.js 16 の ESM module resolution に関する問題。`next/headers` を `next/headers.js` に変更する必要がある可能性があるが、`server-auth.ts` 以外にも同様の import がないか調査が必要。

## 環境

- Next.js 16 + @opennextjs/cloudflare
- Node.js (ESM mode)
- Playwright テストランナー

--- a/.gitignore
+++ b/.gitignore
@@ -19,3 +19,4 @@ a.out
 skills-lock.json
 src/markdown.d.ts
 e2e/_test-import*.spec.ts
+e2e/auth-utils-edge.spec.ts

--- a/e2e/auth-utils.spec.ts
+++ b/e2e/auth-utils.spec.ts
@@ -1,0 +1,24 @@
+import { test, expect } from "@playwright/test";
+import { getJwtExp } from "../src/lib/auth";
+
+test("getJwtExp: 不正なトークンは null を返す", () => {
+  expect(getJwtExp("x.y.z")).toBeNull();
+});
+
+test("getJwtExp: パーツ不足のトークンは null を返す", () => {
+  expect(getJwtExp("onlyonepart")).toBeNull();
+});
+
+test("getJwtExp: 有効な JWT から exp を取得できる", () => {
+  const payload = { sub: "user1", exp: 1234567890 };
+  const encoded = btoa(JSON.stringify(payload)).replace(/=/g, "");
+  const token = `header.${encoded}.signature`;
+  expect(getJwtExp(token)).toBe(1234567890);
+});
+
+test("getJwtExp: exp が数値でない場合は null を返す", () => {
+  const payload = { sub: "user1", exp: "not-a-number" };
+  const encoded = btoa(JSON.stringify(payload)).replace(/=/g, "");
+  const token = `header.${encoded}.signature`;
+  expect(getJwtExp(token)).toBeNull();
+});

--- a/src/lib/auth.ts
+++ b/src/lib/auth.ts
@@ -419,3 +419,17 @@ export async function revokeToken(refreshToken: string): Promise<void> {
     body: JSON.stringify({ refresh_token: refreshToken }),
   });
 }
+
+/** JWT ペイロードの exp クレームを base64 デコードで取得する（署名検証なし） */
+export function getJwtExp(token: string): number | null {
+  try {
+    const parts = token.split(".");
+    if (parts.length < 2) return null;
+    const payload = JSON.parse(atob(parts[1].replace(/-/g, "+").replace(/_/g, "/"))) as {
+      exp?: number;
+    };
+    return typeof payload.exp === "number" ? payload.exp : null;
+  } catch {
+    return null;
+  }
+}

--- a/src/lib/server-auth.ts
+++ b/src/lib/server-auth.ts
@@ -1,12 +1,13 @@
 import { cookies } from "next/headers";
 import { NextResponse } from "next/server";
 import { getCloudflareContext } from "@opennextjs/cloudflare";
-import { verifyJwt, refreshTokens, type RefreshResult } from "./auth";
+import { verifyJwt, refreshTokens, getJwtExp, type RefreshResult } from "./auth";
 import { apiError } from "./api-error";
 import { isCsrfViolation } from "./csrf";
 
 // CSRF 判定ロジックは next/* を含まない形でユニットテスト可能にするため `./csrf` に分離している。
 export { isCsrfViolation } from "./csrf";
+export { getJwtExp } from "./auth";
 
 /**
  * CSRF 違反の場合に 403 NextResponse を、合格時は null を返すラッパー。
@@ -163,20 +164,6 @@ export function deduplicatedRefresh(refreshToken: string): Promise<RefreshResult
     });
   inflightRefresh.set(refreshToken, p);
   return p;
-}
-
-/** JWT ペイロードの exp クレームを base64 デコードで取得する（署名検証なし） */
-export function getJwtExp(token: string): number | null {
-  try {
-    const parts = token.split(".");
-    if (parts.length < 2) return null;
-    const payload = JSON.parse(atob(parts[1].replace(/-/g, "+").replace(/_/g, "/"))) as {
-      exp?: number;
-    };
-    return typeof payload.exp === "number" ? payload.exp : null;
-  } catch {
-    return null;
-  }
 }
 
 /** BETA_ALLOWED_SUBS が設定されている場合、sub がリストに含まれるか確認 */

